XiEditor/EditViewController.swift

```diff
@@ -107,7 +112,7 @@ class EditViewController: NSViewController, EditViewDataSource {
         let first = Int(floor(scrollView.contentView.bounds.origin.y / textMetrics.linespace))
         let height = Int(ceil((scrollView.contentView.bounds.size.height) / textMetrics.linespace))
         let last = first + height
-        if first != firstLine || last != lastLine && document != nil {
```

Comment:
This change was done for a reason (firstLine, lastLine was getting set to the window size, without the update being sent to the core, because document was nil). The change would only be correct if document was guaranteed not to be nil when this is called, which I don't have a lot of confidence about. But if we're sure it's always set, I'm ok with it.